valuable/src/valuable.rs

```diff
@@ -3,11 +3,63 @@ use crate::{Slice, Value, Visit};
 use core::fmt;
 use core::num::Wrapping;
 
+/// A type that can be converted to a [`Value`].
+///
+/// `Valuable` types are inspected by defining a [`Visit`] implementation and
+/// using it whewn calling [`Valuable::visit`]. See [`Visit`] documentation for
+/// more details.
+///
+/// The `Valuable` procedural macro makes implementing `Valuable` easy. Users
+/// can add add `#[derive(Valuable)]` to their types.
```

Comment:
nit: would be nice if this was a link to the derive macro....i can't remember the syntax for disambiguating the link to the macro from the link to the trait of the same name off the top of m yhead...